### PR TITLE
Add multifields support

### LINKED ISSUE #84: in order to deal with multifield, try to get the docvalues field for retrieving the label
see https://github.com/elastic/kibana/pull/7302 for some example

### PR DESCRIPTION
**Pull request by SergeyBondarenko**
Closes #84 

It was tested with the following configuration.

**Mapping**

```
{
  "mappings": {
    "my_city": {
      "properties": {
        "city": {
          "type": "string",
          "fields": {
            "raw": {
              "type":  "string",
              "index": "not_analyzed",
              "doc_values": true
            },
            "rawcity": {
              "type":  "string",
              "index": "not_analyzed",
              "doc_values": true
            },
            "raw_city": {
              "type":  "string",
              "index": "not_analyzed",
              "doc_values": true
            }
          }
        },
        "arrive": {
          "type": "date",
          "fields": {
            "raw_date": {
              "type": "date",
              "doc_values": true
            },
            "raw": {
              "type": "date",
              "doc_values": true
            }
          }
        },
        "depart": {
          "type": "date",
          "fields": {
            "raw": {
              "type": "date",
              "doc_values": true
            }
          }
        }
      }
    }
  }
}
```

**Documents**

```
{
  "city": "New York",
  "arrive": "2016-07-11",
  "depart": "2016-09-05"
},
{
  "city": "York",
  "arrive": "2015-07-11",
  "depart": "2015-09-05"
}

```

### CODE REVIEW

File: public/kibi_timeline_vis_controller.js [hunk 1]
```diff
@@ -70,6 +70,10 @@ define(function (require) {
           $scope.visOptions.groups = [];
           _.each(savedSearchesRes, function ({ savedSearch, groups }, i) {
             for (const group of groups) {
+              // get parent field if multifield
+              // because doc_values feature is not applied to string fields
+              group.labelField = group.labelField.length > 0 ? group.labelField.split('.')[0] : group.labelField;
```
Comment (scampi) [inline public/kibi_timeline_vis_controller.js]: you should use `labelFieldSequence` instead, if available. In the searchsource, you have to add `source` and `fielddata_fields` https://www.elastic.co/guide/en/elasticsearch/reference/2.4/search-request-fielddata-fields.html
This way you can still get the value of the `raw` multifield in kibana.

Comment (SergeyBondarenko) [inline public/kibi_timeline_vis_controller.js]: Do you mean I have to add a new method `fielddata_fields` to the searchsource class here kibi-internal/src/ui/public/courier/data_source/search_source.js ?

**Comment by SergeyBondarenko** [comment]
@scampi Only one test fails - `should init group with dotted field names`. 
I tried to index documents with dots in field names but I don't get an error. How can I run only this particular test without running other tests for the kibi_timileine_viz?

**Comment by szydan** [comment]
@SergeyBondarenko  this failing test might be ok after 
https://github.com/sirensolutions/kibi-internal/pull/2086 gets merged

**Comment by szydan** [comment]
As sirensolutions/kibi-internal#2086 got merged I've rebased issue-84 branch on master
and going to test this PR

File: public/__tests__/kibi_timeline_vis_controller.js [hunk 2]
```diff
@@ -333,18 +333,18 @@ describe('Kibi Timeline', function () {
         expect($scope.visOptions.groups).to.have.length(2);
 
         const expectedA = {
-          labelFieldSequence: [ 'd.o', 't.labelFieldA' ],
-          startFieldSequence: [ 'd.o', 't.startFieldA' ],
-          endFieldSequence: [ 'd.o', 't.endFieldA' ],
+          labelFieldSequence: [ 'd.o.t.labelFieldA' ],
```
Comment (scampi) [inline public/__tests__/kibi_timeline_vis_controller.js]: revert this

File: public/__tests__/kibi_timeline_vis_controller.js [hunk 3]
```diff
@@ -333,18 +333,18 @@ describe('Kibi Timeline', function () {
...
           searchSourceId: '_kibi_timetable_ids_source_flagidAsavedSearchIdA'
         };
         _.assign(expectedA, groups[0]);
         assertGroup($scope.visOptions.groups[0], expectedA);
 
         const expectedB = {
-          labelFieldSequence: [ 'd.o', 't.labelFieldB' ],
-          startFieldSequence: [ 'd.o', 't.startFieldB' ],
-          endFieldSequence: [ 'd.o', 't.endFieldB' ],
+          labelFieldSequence: [ 'd.o.t.labelFieldB' ],
```
Comment (scampi) [inline public/__tests__/kibi_timeline_vis_controller.js]: revert this

File: public/kibi_timeline_vis_controller.js [hunk 4]
```diff
@@ -77,9 +80,12 @@ define(function (require) {
                 searchSource: searchSource,
                 params: {
                   //kibi params
-                  labelFieldSequence: fields[i].byName[group.labelField].path,
-                  startFieldSequence: fields[i].byName[group.startField].path,
-                  endFieldSequence: group.endField && fields[i].byName[group.endField].path || [],
+                  labelFieldSequence: timelineHelper.isMultifield(group.labelField) ? [group.labelField]
```
Comment (scampi) [inline public/kibi_timeline_vis_controller.js]: revert this

File: public/lib/helpers/timeline_helper.js [hunk 5]
```diff
@@ -26,10 +59,15 @@ define(function (require) {
     TimelineHelper.prototype.pluckLabel = function (hit, params, notify) {
       let field;
       if (params.labelFieldSequence) { // in kibi, we have the path property of a field
-        field = kibiUtils.getValuesAtPath(hit._source, params.labelFieldSequence);
```
Comment (scampi) [inline public/lib/helpers/timeline_helper.js]: use logic from https://github.com/elastic/kibana/pull/7302/files#diff-b60fcd2a71f5585fbcbd967a546f5d53R168 to get the values from the `_source` first, and if missing, try to get them from `fields`.

**Pull request closed without merging**